geps/gep-1709.md

```diff
@@ -233,22 +233,22 @@ move on to [certification](#certification) to report the results.
 
 ### Implementation mode
 
-The certification process runs against an implementation using a specific mode
-(e.g., `standard`, `enhanced`) that is defaulted to `standard`. The final
-report will contain such a mode in the modes field. A "mode" is intended to capture
-situations where a Gateway API implementation may have different features and capabilities
-depending on how it is deployed (e.g. an implementation might be deployed with an external,
-OR internal load balancer, and have different capabilities depending on the mode chosen).
-Every certification report `modes` array contains only one element. Reports can be merged
-into one another to provide a single report that accounts for multiple modes, given the same
-Gateway API version.
-omni-comprehensive one, as described [below](#multiple-reports-merge).
+The certification process runs against an implementation using a specific mode.
+The final report will contain such a mode in the modes field. A "mode" is intended
+to capture situations where a Gateway API implementation may have different features
+and capabilities depending on how it is deployed (e.g. an implementation might be
+deployed with an external, OR internal load balancer, and have different capabilities
+depending on the mode chosen). The modes are implementation-specific, and no upstream
+mode is defined, except for `standard`, the default one used in case no mode is specified.
```

Comment:
I'm not 100% sure we should have a default mode. In case an implementation has a unique mode and is not interested in this API functionality, should we just leave the `modes` array empty instead of forcing a `standard` mode?

The more I think about it, the more convinced I get about leaving the array empty and not having a default `standard` mode.

@shaneutt @youngnick any thoughts on this regard?

Reply:
Yeah, I think leaving the array empty by default is fine - but we just need to be clear that if there are multiple results with the same version, they must have different values for the `modes` array. (The same actually goes for having multiple versions with the same mode).

That means that we need to specify the `modes` array as optional with a defined behavior.

If we're going to do this though, we should put the compulsory field (`version`) first, and the optional field (`mode`) second in the docs.

For `mode`, we should use language like:

Implementations MAY have multiple entries for the same `version` if and only if each entry has a distinct `modes` array. Having two results with the same `version` and identical `modes` arrays is invalid.

Reply:
If we go with my suggestion below, I'd recommend that mode is not a list and just a single `string`, and we default to the Gateway Class Name used in the test run as `mode` unless a different value is specified.